### PR TITLE
fix clippy::single_char_pattern perf findings

### PR DESCRIPTION
**Pull request by matthiaskrgr**

### CODE REVIEW

File: compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs [hunk 1]
```diff
@@ -741,7 +741,7 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
 
                     let msg = format!(
                         "the trait bound `{}: {}` is not satisfied",
-                        orig_ty.to_string(),
+                        orig_ty,
```
Comment (jyn514) [inline compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs:744]: This one is fixing a different issue, but that's fine, it's also a good change :)

File: src/librustdoc/html/render/cache.rs [hunk 2]
```diff
@@ -184,8 +184,8 @@ crate fn build_index<'tcx>(krate: &clean::Crate, cache: &mut Cache, tcx: TyCtxt<
         })
         .expect("failed serde conversion")
         // All these `replace` calls are because we have to go through JS string for JSON content.
-        .replace(r"\", r"\\")
-        .replace("'", r"\'")
+        .replace('\\', r"\\")
+        .replace('\'', r"\'")
```
Comment (jyn514) [inline src/librustdoc/html/render/cache.rs:188]: Hmm, I think was more clear before - can you change both of these to raw string?

**Review by jyn514 — APPROVED** [review]
r=me with the nit fixed :)

File: compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs (hunk 1 above)
Comment (matthiaskrgr) [reply compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs:744]: nothing to see here, please move along :p

**Pull request merged**